apps/arweave/src/ar_data_sync.erl

```diff
@@ -981,68 +1017,115 @@ handle_cast(sync_data2, State) ->
 %%    b. Sync queue is busy. Will pause until previously queued intervals are scheduled to the
 %%       ar_data_sync_worker_master for syncing.
 handle_cast(collect_peer_intervals, State) ->
-	#sync_data_state{ range_start = Start, range_end = End } = State,
+	#sync_data_state{ range_start = Start, range_end = End,
+			disk_pool_threshold = DiskPoolThreshold,
+			sync_phase = SyncPhase,
+			store_id = StoreID,
+			migrations_index = MI } = State,
 	?LOG_DEBUG([{event, collect_peer_intervals_start},
 		{function, collect_peer_intervals},
 		{store_id, State#sync_data_state.store_id},
 		{s, Start}, {e, End}]),
-	gen_server:cast(self(), {collect_peer_intervals, Start, End}),
-	{noreply, State};
+	CheckIsJoined =
+		case ar_node:is_joined() of
+			false ->
+				ar_util:cast_after(1000, self(), collect_peer_intervals),
+				false;
+			true ->
+				true
+		end,
+	IsFootprintRecordMigrated =
+		case ar_kv:get(MI, ?FOOTPRINT_MIGRATION_CURSOR_KEY) of
+			{ok, <<"complete">>} ->
+				true;
+			_ ->
+				ar_util:cast_after(5_000, self(), collect_peer_intervals),
+				false
+		end,
+	IntersectsDiskPool =
+		case CheckIsJoined andalso IsFootprintRecordMigrated of
+			false ->
+				noop;
+			true ->
+				End > DiskPoolThreshold
+		end,
+	%% Alternate between "normal" and footprint-based syncing.
+	%% Footprint-based syncing downloads replica 2.9 chunks footprint by footprint
+	%% to avoid redundant entropy generations for unpacking. "Normal" syncing ignores
+	%% replica 2.9 data and mostly downloads unpacked data from peers storing it.
+	SyncPhase2 =
```

Comment:
is this the basic stucture:
1. Collect peer intervals for the range `range_start` to `range_end` for all peers which advertise data in that range
2. first time through we sync "normally"
3. Everytime we complete syncing the `range_start` to `range_end` range, we start over and flip the move from `normal` to `footprint` or back

so we do a full pass through the range in one mode, then a full pass in the other mode, etc...